tests/system-tests/rdscore/internal/rdscorecommon/egress-ip.go

```diff
@@ -0,0 +1,1018 @@
+
+				if err != nil {
+					glog.V(100).Infof("Failed to parse %q for host/port pair", result.String())
+
+					return false, nil
+				}
+
+				glog.V(100).Infof("Verify IP version type correctness")
+
+				myIP, err := netip.ParseAddr(parsedIP)
+
+				if err != nil {
+					glog.V(100).Infof("Failed to parse used ip address %q", parsedIP)
+
+					return false, nil
+				}
+
+				if isIPv6 && myIP.Is4() || !isIPv6 && myIP.Is6() {
+					glog.V(100).Infof("Wrong IP version detected; %q", parsedIP)
+
+					return false, nil
+				}
+
+				glog.V(100).Infof("Comparing %q with expected %q", parsedIP, expectedIPs)
+
+				for _, expectedIP := range expectedIPs {
+					if parsedIP == expectedIP {
+						return true, nil
+					}
+				}
+
+				glog.V(100).Infof("Mismatched IP address. Expected %q got %q", expectedIPs, parsedIP)
+
+				return false, nil
+			})
+
+		if err != nil {
+			return fmt.Errorf("failed to run command from within pod %s: %w", clientPod.Object.Name, err)
+		}
+	}
+
+	return nil
+}
+
+func getEgressIPMap() (map[string]string, error) {
+	By("Getting a map of source nodes and assigned Egress IPs for these nodes")
+
+	egressIPObj, err := egressip.Pull(APIClient, RDSCoreConfig.EgressIPName)
+
+	if err != nil {
+		glog.V(100).Infof("Failed to pull egressIP %q object: %v", RDSCoreConfig.EgressIPName, err)
+
+		return nil, fmt.Errorf("failed to retrieve egressIP %s object: %w", RDSCoreConfig.EgressIPName, err)
+	}
+
+	egressIPMap, err := egressIPObj.GetAssignedEgressIPMap()
+
+	if err != nil {
+		glog.V(100).Infof("Failed to retrieve egressIP %s assigned egressIPs map: %v",
+			RDSCoreConfig.EgressIPName, err)
+
+		return nil, fmt.Errorf("failed to retrieve egressIP %s assigned egressIPs map: %w",
+			RDSCoreConfig.EgressIPName, err)
+	}
+
+	if len(egressIPMap) == 0 {
+		return nil, fmt.Errorf("egressIPs configuration failure, no assigned to the nodes ips were found: %v",
+			egressIPMap)
+	}
+
+	return egressIPMap, nil
+}
+
+func getEgressIPList() ([]string, error) {
+	By("Build an egressIP list")
+
+	var eIPList []string
+
+	egressIPMap, err := getEgressIPMap()
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve egressIP map due to %w", err)
+	}
+
+	for eIPNodeName, eIPValue := range egressIPMap {
+		if eIPValue != RDSCoreConfig.EgressIPv4 && eIPValue != RDSCoreConfig.EgressIPv6 {
+			return nil, fmt.Errorf("EgressIP address assigned to the node %s not correctly configured; "+
+				"current value: %s, expected values to be set %s or %s", eIPNodeName, eIPValue,
+				RDSCoreConfig.EgressIPv4, RDSCoreConfig.EgressIPv6)
+		}
+
+		eIPList = append(eIPList, eIPValue)
+	}
+
+	return eIPList, nil
+}
+
+func verifyEgressIPConnectivityBalancedTraffic(isIPv6 bool) {
+	By("Verifying egress IP connectivity balanced traffic; single namespace")
+
+	expectedIPs, err := getEgressIPList()
+	Expect(err).ToNot(HaveOccurred(),
+		fmt.Sprintf("Failed to retrieve configured eIP addresses list from the egressIP %s: %v",
```

Comment:
let use full name to avoid ambiguity 
```suggestion
		fmt.Sprintf("Failed to retrieve configured EgressIP addresses list from the egressIP %s: %v",
```